esx_service/vmdk_ops.py

```diff
@@ -490,24 +492,33 @@ def get_vol_path(datastore, tenant_name=None):
     if os.path.isdir(path):
         # If the path exists then return it as is.
         logging.debug("Found %s, returning", path)
-        return path
+        return path, None
 
     if not os.path.isdir(dock_vol_path):
         # The osfs tools are usable for DOCK_VOLS_DIR on all datastores
         cmd = "{0} {1}".format(OSFS_MKDIR_CMD, dock_vol_path)
         rc, out = RunCommand(cmd)
-        if rc != 0:
-            logging.warning("Failed to create %s", dock_vol_path)
-            return None
+        errMsg = None
+        if rc == ERR_OSFS_MKDIR_READONLY:
+            errMsg = "{0} creation failed - {1} not writable".format(DOCK_VOLS_DIR, datastore)
+        elif rc == ERR_OSFS_MKDIR_NOPERMISSION:
+            errMsg = "{0} creation failed - Permission denied on {1}".format(DOCK_VOLS_DIR, datastore)
+        else:
+            logging.warning("Failed to initialize volume path %s", path)
+            errMsg = "Failed to initialize volume path {0}".format(path)
```

Comment:
won't it hit the 'else' clause and print this error message even when RunCommand was successful ?  I think you need to have 
```python
  elif rc == 0:
    pass
```
somewhere